### ISSUE TITLE
Bug: Top-Level library targets not being merged correctly

### ISSUE THREAD

**Issue by sebastianv1** [buildbuddy-io/rules_xcodeproj#668]
## Describe the bug
Top-level library targets used as the dependency for an `ios_application` that also have a sibling test target/library **without** a test host end up being marked as a non-mergable target. This manifests downstream as our primary `ios_application` target inherits the `DEBUG_INFORMATION_FORMAT` build setting as DWARF with dSYM File from Xcode's iOS Default instead of being set as just DWARF from the library target.

When this build setting is configured, the generate dSYM step in Xcode also generates warnings for every swiftmodule it tries to link into the dSYM bundle. Seems like it just can't find the swiftmodule path and runs `stat` over a missing file? Hence the timestamp date being 12/31/1969 😅 
```
/.../Example/Example.swiftmodule/arm64-apple-ios-simulator.swiftmodule: timestamp mismatch between swift interface file (1969-12-31 19:00:00.000000000) and debug map (1969-12-31 19:00:00.000000000)
```

Additionally, the malformed dYSM causes `p` / `po` break in lldb even with the remaps (`v` still works though, hence the missing swiftmodule ast paths)
`error: Couldn't realize type of self`

It appears there are 2 distinct but connected issues here:
1. Top-level targets used by an `ios_application` that are also a dependency for an `ios_unit_test/swift_library` duo end up in the `non_mergable_targets` because removing the `test_host` doesn't de-dupe the linked static libraries list. Why do we use this list as the input for determining mergeable targets?
2. Is dSYM generation by Xcode currently supported in rules_xcodeproj?

## Reproduction steps
Removing the test host for `ExampleTests` in `examples/ios_app` reproduces this issue in the generated project.

## Expected behavior
Skip Xcode dSYM generation when not requested by the Bazel Build.

## Versions
- Bazel: `6.0.0-pre.20220310.1`
- rules_xcodeproj: `0.5.0`
- rules_apple: `1.0.1`
- rules_swift: `1.0.0`

**Comment by sebastianv1** [comment]
cc: @cgrindel

**Comment by brentleyjones** [comment]
This is because a static library can't merge into two targets. Target merging is to replicate the Xcode setup of code being compiled as part of a top-level target. If that code is used in two places then it's actually a static library, not the binary of a top-level target.

That said, I'll look into the repro steps and see how this situation can be improved. The warning most likely is expected though.

**Comment by brentleyjones** [comment]
dSYM generation should be supported, but we don't have a test for it, so it might have broke.

**Comment by brentleyjones** [comment]
https://github.com/buildbuddy-io/rules_xcodeproj/pull/669 fixes the `DEBUG_INFORMATION_FORMAT` setting. I need to add a test for dSYM generation next.

**Comment by brentleyjones** [comment]
>When this build setting is configured, the generate dSYM step in Xcode also generates warnings for every swiftmodule it tries to link into the dSYM bundle. Seems like it just can't find the swiftmodule path and runs stat over a missing file? Hence the timestamp date being 12/31/1969

I only get that warning when using lld. Do you have `--ld-path` set? The `ios_app` example does.

**Comment by brentleyjones** [comment]
After #669 and #670 are merged, I think we can close this issue. Target merging is working as expected, as the app and the test both want to link to the static library, so it can't merge into the app. In an Xcode-only world this common library would be a dependency of the app library. Also, an unmerged library isn't the end of the world, and I should probably adjust the warning criteria in the future.
